addons/mail/models/mail_mail.py

```diff
@@ -78,9 +78,19 @@ def default_get(self, fields):
     auto_delete = fields.Boolean(
         'Auto Delete',
         help="This option permanently removes any track of email after it's been sent, including from the Technical menu in the Settings, in order to preserve storage space of your Odoo database.")
+    to_delete = fields.Boolean('To Delete', help='Define if the mail will be removed in the autovacuum CRON')
     scheduled_date = fields.Char('Scheduled Send Date',
         help="If set, the queue manager will send the email after the date. If not set, the email will be send as soon as possible.")
 
+    def init(self):
+        """Create a partial index on "to_delete" to make the search on those records fast.
+
+        The benefit on this partial index is to not have a big impact on the
+        update / insert of other records in the database in comparison to a standard
+        index.
+        """
+        self._cr.execute("CREATE INDEX ON mail_mail(id) WHERE to_delete=TRUE;")
```

Comment:
Created indexes in the init method usually call:
`CREATE INDEX IF NOT EXISTS mail_mail_to_delete_idx [...]`

I'm guessing it's better to support this method being called multiple times potentially?

Side-note: `WHERE to_delete=TRUE;` -> `WHERE to_delete = TRUE;` ?